app/Services/Email.php

```diff
@@ -70,10 +70,10 @@ protected function defineTokens($user)
         // @TODO - set defaults?
         $tokens = [
             ':campaign_title:'        => $this->contest->campaign->title,
-            ':end_date:'              => $this->competition->competition_end_date->format('F d, Y'),
+            ':end_date:'              => isset($this->competition) ? $this->competition->competition_end_date->format('F d, Y') : '',
```

Comment:
shouldn't you just check if this is null here? Since you set it as `null` in the constructor?